tailcall-prettier/src/lib.rs

```diff
@@ -0,0 +1,80 @@
+use std::fmt::{Display, Formatter};
+use std::io::Write;
+use std::process::{Command, Stdio};
+
+use anyhow::{anyhow, Result};
+
+enum FileTypes {
+    Gql,
+    Yml,
+    Json,
+    Md,
+    Ts,
+    Js,
+}
+
+impl Display for FileTypes {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        match self {
+            FileTypes::Gql => write!(f, "graphql"),
+            FileTypes::Yml => write!(f, "yml"),
+            FileTypes::Json => write!(f, "json"),
+            FileTypes::Md => write!(f, "md"),
+            FileTypes::Ts => write!(f, "ts"),
+            FileTypes::Js => write!(f, "js"),
+        }
+    }
+}
+
+impl FileTypes {
+    fn detect(path: &str) -> Result<Self> {
+        let ext = path
+            .split('.')
+            .last()
+            .ok_or(anyhow!("No file extension found"))?
+            .to_lowercase();
+        match ext.as_str() {
+            "gql" | "graphql" => Ok(FileTypes::Gql),
+            "yml" | "yaml" => Ok(FileTypes::Yml),
+            "json" => Ok(FileTypes::Json),
+            "md" => Ok(FileTypes::Md),
+            "ts" => Ok(FileTypes::Ts),
+            "js" => Ok(FileTypes::Js),
+            _ => Err(anyhow!("Unsupported file type")),
+        }
+    }
+}
+
+pub fn format_with_prettier<T: AsRef<str>>(code: T, file_ty: T) -> Result<String> {
```

Comment:
Keep the API close to — 
https://github.com/prettier/prettier/blob/main/src/index.js#L62C29-L62C36

Reply:
```suggestion
pub fn format_with_prettier<T: AsRef<str>>(code: T, parser: Parser) -> Result<String> {
```